### PR TITLE
Resolve StoredCredential.Descriptor at access(?)

### PR DESCRIPTION
**Pull request by abergs**
While we might want to cache it, this is to simplify the concept so that devs more easier understand how the piecies fit toghether"

### CODE REVIEW

File: Src/Fido2.Development/StoredCredential.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public class StoredCredential
 
     public byte[] UserId { get; set; }
 
-    public PublicKeyCredentialDescriptor Descriptor { get; set; }
```
Comment (abergs) [inline Src/Fido2.Development/StoredCredential.cs:55]: For less allocations we could cache the instance of the PublicKeyCredentialDescriptor as a private member.

File: Src/Fido2.Development/StoredCredential.cs [hunk 2]
```diff
@@ -52,7 +52,7 @@ public class StoredCredential
...
+    public PublicKeyCredentialDescriptor Descriptor => new(PublicKeyCredentialType.PublicKey, Id, Transports);
```
Comment (abergs) [inline Src/Fido2.Development/StoredCredential.cs:55]: For less allocations we could cache the instance of the PublicKeyCredentialDescriptor as a private member.

**Comment by abergs** [comment]
@Regenhardt Does this perhaps strike a good balance on your original concerns in #510

**Comment by Regenhardt** [comment]
Thanks for the ping, yes this is a nice solution for the redundancy. Could use an XML summary on the descriptor and maybe on the Id mentioning it as being the actual public key or something.

File: Src/Fido2.Development/StoredCredential.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@ public class StoredCredential
     /// <summary>
     /// The Credential ID of the public key credential source.
     /// </summary>
-    public byte[] Id { get; set; }
+    public required byte[] Id { get; set; }
```
Comment (iamcarbon) [inline Src/Fido2.Development/StoredCredential.cs:12]: Very happy to see we're on .NET8, and are able to use a required property here.

Comment (iamcarbon) [reply Src/Fido2.Development/StoredCredential.cs:12]: Can this also be made immutable with { get; init; }?

File: Src/Fido2.Development/StoredCredential.cs (hunk 2 above)
Comment (iamcarbon) [reply Src/Fido2.Development/StoredCredential.cs:55]: This looks good as-is to me. The consumer is likely to capture this into a variable, and it's a tiny object.

File: Src/Fido2.Development/StoredCredential.cs (hunk 3 above)
Comment (abergs) [reply Src/Fido2.Development/StoredCredential.cs:12]: @iamcarbon It probably could, but that opens the question if not more of these should be made requried/init-only etc so in an effort to merge this I won't add it.

Also - I need to remind myself that this type is only used in the Development namespace, so it's just used for demoes etc.d

**Pull request merged**